### PR TITLE
[CELEBORN-576] Add static identity provider and manually settable identity provider for non-hadoop environment.

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
1. Add DefaultIdentityProvider to eliminate Hadoop dependency.
2. Add manually settable IdentityProvider to let the users decide their own tenant and username.
3. Rename the previous DefaultIdentityProvider to HadoopBasedIdentityProvider.

### Why are the changes needed?
Ditto.

### Does this PR introduce _any_ user-facing change?
Yes, config's default value has been changed.

### How was this patch tested?
UT.

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/identity/UserSpecifiedIdentity.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.identity
+
+import org.apache.hadoop.security.UserGroupInformation
+
+import org.apache.celeborn.common.CelebornConf
+
+class UserSpecifiedIdentity extends IdentityProvider {
```
[the remark this answers is not in the document]
Comment (FMX) [reply common/src/main/scala/org/apache/celeborn/common/identity/UserSpecifiedIdentity.scala]: Yes, you are right. The class name changed.

File: common/src/main/scala/org/apache/celeborn/common/identity/UserSpecifiedIdentityProvider.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.identity
+
+import org.apache.hadoop.security.UserGroupInformation
+
+import org.apache.celeborn.common.CelebornConf
+
+class UserSpecifiedIdentityProvider extends IdentityProvider {
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/identity/UserSpecifiedIdentityProvider.scala:24]: CustimizedIdentityProvider？

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 3]
```diff
@@ -2647,12 +2647,31 @@ object CelebornConf extends Logging {
...
+        s"Optional values: " +
+        s"org.apache.celeborn.common.identity.HadoopBasedIdentityProvider user name will be obtained by UserGroupInformation.getUserName; " +
+        s"org.apache.celeborn.common.identity.UserSpecifiedIdentityProvider user name and tenant id will be set manually;" +
+        s"org.apache.celeborn.common.identity.DefaultIdentityProvider uesr name and tenant id are default values.")
       .version("0.2.0")
       .stringConf
       .createWithDefault(classOf[DefaultIdentityProvider].getName)
 
+  val QUOTA_SPECIFIED_TENANT: ConfigEntry[String] =
+    buildConf("celeborn.quota.identity.specified.tenant")
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2660]: `celeborn.quota.identity.customized.tenant`?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 4]
```diff
@@ -2647,12 +2647,31 @@ object CelebornConf extends Logging {
...
+      .categories("quota")
+      .doc(s"Tenant id if celeborn.quota.identity.provider is org.apache.celeborn.common.identity.UserSpecifiedIdentity.")
+      .version("0.3.0")
+      .stringConf
+      .createWithDefault(IdentityProvider.DEFAULT_TENANT_ID)
+
+  val QUOTA_SPECIFIED_USERNAME: ConfigEntry[String] =
+    buildConf("celeborn.quota.identity.specified.userName")
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2668]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 5]
```diff
@@ -2647,12 +2647,30 @@ object CelebornConf extends Logging {
...
+        s"org.apache.celeborn.common.identity.HadoopBasedIdentityProvider user name will be obtained by UserGroupInformation.getUserName; " +
+        s"org.apache.celeborn.common.identity.DefaultIdentityProvider uesr name and tenant id are default values or user-specific values.")
       .version("0.2.0")
       .stringConf
       .createWithDefault(classOf[DefaultIdentityProvider].getName)
 
+  val QUOTA_USER_SPECIFIC_TENANT: ConfigEntry[String] =
+    buildConf("celeborn.quota.identity.user-specific.tenant")
+      .categories("quota")
+      .doc(s"Tenant id if celeborn.quota.identity.provider is org.apache.celeborn.common.identity.UserSpecifiedIdentity.")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2661]: UserSpecifiedIdentity -> DefaultIdentityProvider

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 6]
```diff
@@ -2647,12 +2647,30 @@ object CelebornConf extends Logging {
...
+      .version("0.3.0")
+      .stringConf
+      .createWithDefault(IdentityProvider.DEFAULT_TENANT_ID)
+
+  val QUOTA_USER_SPECIFIC_USERNAME: ConfigEntry[String] =
+    buildConf("celeborn.quota.identity.user-specific.userName")
+      .categories("quota")
+      .doc(s"User name if celeborn.quota.identity.provider is org.apache.celeborn.common.identity.UserSpecifiedIdentity.")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2669]: ditto

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**